### PR TITLE
feat(ui): Dark Mode

### PR DESCRIPTION
**Pull request by Hazer**
## Description
Includes dark mode, auto selects and allows for manually overriding with user preference

### Screenshot
Screenshot 2024-05-02 at 09 30 50
Screenshot 2024-05-02 at 09 31 48
Screenshot 2024-05-02 at 09 32 04
Screenshot 2024-05-02 at 09 32 15
Screenshot 2024-05-02 at 09 31 23
Screenshot 2024-05-02 at 09 31 33

### Depends on:
- Merging #2491

## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

**Comment by Hazer** [comment]
I didn't ask before, but any considerations regarding colors? Specially with the NavBar colors?

**Comment by ReenigneArcher** [comment]
> I didn't ask before, but any considerations regarding colors? Specially with the NavBar colors?

The white(ish) on yellow(ish) is probably going to be difficult to read. I think I'd prefer the same navbar colors for both, e.g. the original. But everywhere else seems to look okay from your screenshots.

Also, I apologize for the delays on getting all these reviewed and in, I'm a bit overwhelmed with this project at the moment, having 50 open PRs. Are you on our discord? If so it would probably easier to communicate about any of these and get them across the goal line.

**Comment by Hazer** [comment]
Draft until #2491 is merged.

**Comment by Hazer** [comment]
Screenshot 2024-05-13 at 15 40 31
Screenshot 2024-05-13 at 15 40 43

Dark mode navbar colors updated

File: src_assets/common/assets/web/public/assets/locale/en.json [hunk 1]
```diff
@@ -334,7 +334,11 @@
     "home": "Home",
     "password": "Change Password",
     "pin": "Pin",
-    "troubleshoot": "Troubleshooting"
+    "troubleshoot": "Troubleshooting",
+    "toggle_theme": "Toggle theme",
+    "theme_auto": "Auto",
+    "theme_light": "Light",
+    "theme_dark": "Dark"
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/public/assets/locale/en.json:341]: Could you make these alphabetical?

File: src_assets/common/assets/web/Navbar.vue [hunk 2]
```diff
@@ -1,60 +1,93 @@
...
+    discordWidget.setAttribute('src', 'https://app.lizardbyte.dev/js/discord.js')
+    document.head.appendChild(discordWidget)
+  }
 }
 </script>
 
 <style>
-.nav-link.active {
-    font-weight: 500;
+.navbar-background {
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/Navbar.vue:63]: Would it make sense to separate out all the styles into the sunshine.css file?

File: src_assets/common/assets/web/Navbar.vue [hunk 3]
```diff
@@ -1,60 +1,93 @@
...
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="/password"><i class="fas fa-fw fa-user-shield"></i> {{
+                $t('navbar.password')
+              }}</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="/troubleshooting"><i class="fas fa-fw fa-info"></i> {{
+                $t('navbar.troubleshoot')
+              }}</a>
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/Navbar.vue:33]:
```suggestion
            <a class="nav-link" href="/password"><i class="fas fa-fw fa-user-shield"></i>
                {{ $t('navbar.password') }}</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" href="/troubleshooting"><i class="fas fa-fw fa-info"></i>
                {{ $t('navbar.troubleshoot') }}</a>
```

Having the i18n opening/closing brackets on separate lines is a bit confusing.

**Review by ReenigneArcher — CHANGES_REQUESTED** [review]

File: src_assets/common/assets/web/Navbar.vue (hunk 3 above)
Comment (Hazer) [reply src_assets/common/assets/web/Navbar.vue:33]: CLion reformat got me there, I wasn't expecting this haha

File: src_assets/common/assets/web/Navbar.vue (hunk 2 above)
Comment (Hazer) [reply src_assets/common/assets/web/Navbar.vue:63]: I'd rather have each component with it's own style, with time I think it makes things easier, but I was thinking how could I move all the default colors, text sizes and stuff to sunshine.css, kind of creating a design system or theme , but in another PR

File: src_assets/common/assets/web/public/assets/locale/en.json (hunk 1 above)
Comment (Hazer) [reply src_assets/common/assets/web/public/assets/locale/en.json:341]:
```suggestion
    "theme_auto": "Auto",
    "theme_dark": "Dark",
    "theme_light": "Light",
    "toggle_theme": "Toggle theme",
    "troubleshoot": "Troubleshooting"
```

File: src_assets/common/assets/web/Navbar.vue [hunk 4]
```diff
@@ -1,60 +1,93 @@
...
+          <li class="nav-item">
+            <a class="nav-link" href="/apps"><i class="fas fa-fw fa-stream"></i> {{ $t('navbar.applications') }}</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="/config"><i class="fas fa-fw fa-cog"></i> {{ $t('navbar.configuration') }}</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" href="/password"><i class="fas fa-fw fa-user-shield"></i> {{
+                $t('navbar.password')
+              }}</a>
```
Comment (Hazer) [inline src_assets/common/assets/web/Navbar.vue:28]:
```suggestion
            <a class="nav-link" href="/password"><i class="fas fa-fw fa-user-shield"></i> {{ $t('navbar.password') }}</a>
```

File: src_assets/common/assets/web/Navbar.vue (hunk 3 above)
Comment (Hazer) [inline src_assets/common/assets/web/Navbar.vue:33]:
```suggestion
            <a class="nav-link" href="/troubleshooting"><i class="fas fa-fw fa-info"></i> {{ $t('navbar.troubleshoot') }}</a>
```

**Review by ReenigneArcher — APPROVED** [review]

**Comment by ReenigneArcher** [comment]
I don't know why, but the toggle drop down doesn't work on the "Applications" tab.

File: src_assets/common/assets/web/public/assets/locale/en.json [hunk 5]
```diff
@@ -334,6 +334,10 @@
     "home": "Home",
     "password": "Change Password",
     "pin": "Pin",
+    "theme_auto": "Auto",
+    "theme_dark": "Dark",
+    "theme_light": "Light",
+    "toggle_theme": "Toggle theme",
```
Comment (ReenigneArcher) [inline src_assets/common/assets/web/public/assets/locale/en.json:340]: How about just using "Theme"?

Comment (Hazer) [reply src_assets/common/assets/web/public/assets/locale/en.json:340]:
```suggestion
    "toggle_theme": "Theme",
```

**Comment by Hazer** [comment]
> I don't know why, but the toggle drop down doesn't work on the "Applications" tab.

That's odd. I have really no idea why too, there's no obvious reason for it, I'll need to debug later

**Comment by ReenigneArcher** [comment]
> > I don't know why, but the toggle drop down doesn't work on the "Applications" tab.
> 
> That's odd. I have really no idea why too, there's no obvious reason for it, I'll need to debug later

@TheElixZammuto any idea what could cause this?

The "button" works on every other navbar page, and there are no errors in the browser console.

**Comment by Hazer** [comment]
> The "button" works on every other navbar page, and there are no errors in the browser console.

Fixed. It had to do with bootstrap module bundling being implicitly duplicated. It'll probably be a good idea to revamp the vite configs in the future, for now I just removed the duplication.

ref:
https://github.com/aurelia/skeleton-navigation/issues/894
https://discourse.aurelia.io/t/bootstrap-import-bootstrap-breaks-dropdown-menu-in-navbar/641/9

**Review by ReenigneArcher — APPROVED** [review]

**Comment by karasuhebi** [comment]
@Hazer Thanks for this!

**Pull request merged**